### PR TITLE
*: fix nakedret lint

### PR DESCRIPTION
**Pull request by fuweid**
Part of #15549

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @fuweid 👍🏻

**Comment by ahrtr** [comment]
Almost all the named return parameters are bad examples. The only exception is the one used in wal.go, it makes sense, because it makes the returned value more clear.

So suggest to convert all the named return parameter to naked return parameters except the wal.go. 

Reference: https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

**Comment by fuweid** [comment]
@ahrtr sorry for the late reply. The nakedret is used to detect the named return lint issue. The code change is caused by nakedret. I think we can keep the named return var for `retErr` check in defer or interface documentation. In this patch, you can see SelfCert is long function. Multiple naked return branches are far away from the declaration.  IMO, it's hard to review the code when there is any change. And even if err isn't nil, the function still return non-empty value. Even though the func-invoker should not use the non-empty value if the err isn't nil, it's still confuse.

Maybe we can improve the https://github.com/alexkohler/nakedret/blob/564f756361f1c6a648f759199ab7c03c59cae6dc/cmd/nakedret/main.go#L16 to config max function length which the nakedret tolerances. Or if it's not useful in etcd repo, we can consider to remove the lint.

cc @jmhbnz @serathius

File: client/pkg/transport/keepalive_listener.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ type tlsKeepaliveListener struct {
 func (l *tlsKeepaliveListener) Accept() (c net.Conn, err error) {
```
Comment (ahrtr) [inline client/pkg/transport/keepalive_listener.go:104]:
```suggestion
func (l *tlsKeepaliveListener) Accept() (net.Conn, error) {
```

**Comment by ahrtr** [comment]
To be clearer, the PR makes sense to me. I was requesting to make change something like https://github.com/etcd-io/etcd/pull/16601#discussion_r1328108604. The only exception is the named return parameter used in wal.go, it makes sense, so let's keep it as it's.

**Comment by ahrtr** [comment]
Sorry for the confusion, I should be more clearer in the first place.

File: server/etcdserver/bootstrap_test.go [hunk 2]
```diff
@@ -245,7 +250,7 @@ func prepareData(cfg config.ServerConfig) (err error) {
 func createWALFileWithSnapshotRecord(cfg config.ServerConfig, snapshotTerm, snapshotIndex uint64) (err error) {
```
Comment (fuweid) [inline server/etcdserver/bootstrap_test.go:250]: The `err` is used in defer so keep it here.

Comment (ahrtr) [reply server/etcdserver/bootstrap_test.go:250]: ack

**Review by ahrtr — APPROVED** [review]
LGTM

thx @fuweid

**Review by serathius — APPROVED** [review]

**Pull request merged**